### ISSUE TITLE
minor suggestions

### ISSUE THREAD

**Issue by soerenwolfers** [ywelsch/duckdb-psql#4]
First of all, great job with this! I wrote a feature request at the main duckdb repo asking almost for exactly what I just now found out you've already done (see  https://github.com/duckdb/duckdb/discussions/14187)

Would you be open to 
- Change the pipe character to `|>` to avoid clashes with the bitwise OR (`|>` is already used in this manner in a number of functional languages)
- Rewrite to a nested FROM clause instead of a CTE, to avoid clashes with the autogenerated query aliases
?

I'm going to recommend this extension wherever I can in the hope it can gain traction with more people and eventually be fleshed out to something that doesn't need the special syntax for subqueries anymore either.

**Comment by ywelsch** [comment]
Thanks for the feedback. Changing the pipe character looks reasonable to me. The nested FROM clauses are also a great idea. I will try to pick this up.

Ideally this extension here would be reworked by making changes to the DuckDB parser directly (and embedding a modified copy of the parser in the extension).

**Comment by soerenwolfers** [comment]
yes, I also wonder whether it's worth rewriting anything before hooking this into the proper parser. 

btw, did you see the Google paper that edited into my issue? it seems there is s common sentiment that SQL needs to evolve this way ;)

**Comment by ywelsch** [comment]
I've added the above changes (changing pipe character and using nested from clauses). I've also made this a DuckDB community extension, to make installation simpler. Try it out with `install psql from community;` in DuckDB 1.1.1 and let me know how it goes.

Reworking this to use the proper parser will take more time. There is also upcoming work in DuckDB itself to move to a different parser (and I would like to avoid repeating that work in a new parser infrastructure).

I was / am aware of the Google paper. My original attempts here predated that paper (or at least my knowledge thereof). DuckDB has other cool SQL features that makes its query language even more powerful (no need e.g. for some of the other extra operators introduced in Google's paper as DuckDB already has powerful syntax to do things like `SELECT * EXCEPT X`).

I hope that further down the road SQL will see some standardization around some of these more powerful language features. In the meantime we can build the excitement around that with extensions like this one here.

Thanks for all the good feedback here, it gave me the motivation to move this extension one step further (e.g. make it a community extension).
